lib/ts/IntStr.h

```diff
@@ -0,0 +1,100 @@
+/** @file
+
+    Class for efficient conversion from signed and unsigned 64-bit integers to strings.
+
+    @section license License
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+#include <ts/string_view.h>
+
+namespace ts
+{
+class IntStr
+{
+public:
+  static const std::size_t MaxSize = 20;
+
+  IntStr(std::uint64_t v) { _gen(v); }
+
+  IntStr(std::int64_t v)
+  {
+    if (v >= 0) {
+      _gen(v);
+
+    } else {
+      _gen(static_cast<std::uint64_t>(-v));
+
+      _buf[MaxSize - (++_size)] = '-';
+    }
+  }
+
+  const char *
+  data() const
+  {
+    return _buf + MaxSize - _size;
+  }
+
+  std::size_t
+  size() const
+  {
+    return _size;
+  }
+
+  ts::string_view
+  sV() const
```

Comment:
I would suggest a better name would be to use as_xyz() for objects with conversions functions to a different object.
so something like:
 as_string_view()
 as_sv()
 asStringView()

would be a better convention to use on all our objects

Reply:
I think short abbreviations have their place, but I already lost that argument.  If it's going to be long anyway, we could just have the conversion operator, and write ts::string_view(intStr) when we want it to be explicit.  Although @zwoop may insist it be static_cast<ts::string_view>(intStr) .

Reply:
I am just looking for clarity in the functions. I am for names that say intent. asType() is a common pattern that makes it clear what the intent is in the function, which makes it easy to understand when reading. sV() is unclear. Explicit conversion via static_cast or a clearly named function, I am ok with both I would rather have the code read cleaning. I think this I small and a good improvement to the code.